src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int SALARY_INDEX = 2;
+    private static final int HOURS_INDEX = 3;
+    private static final DateTimeFormatter FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private int salaryAmount;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder()
+                .append("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            salaryAmount = 0;
+            builder.append(System.lineSeparator())
+                    .append(names[i]).append(" - ");
+            for (int j = 0; j < data.length; j++) {
+                if (names[i].equals(data[j].split(" ")[NAME_INDEX])) {
```

Comment:
let's create an array and split `data` into it to not split every time we need to access some value from it